app/controllers/ems_common.rb

```diff
@@ -165,6 +165,39 @@ def edit
                     :url  => "/#{controller_name}/#{@ems.id}/edit")
   end
 
+  def properties
+    # Get the query parameters:
+    id = params[:id]
+    type = params[:type]
+
+    # Assert the permissions:
+    permission_suffix = if id.blank? then :new else :edit end
+    assert_privileges("#{permission_prefix}_#{permission_suffix}")
+
+    # Check the id:
+    if id.blank?
+      @ems = model.new
+    else
+      begin
+        @ems = find_record_with_rbac(model, id)
+      rescue => err
+        head :not_found
+        return
+      end
+    end
+
+    # Check the type:
+    unless ExtManagementSystem.supported_types.include?(type)
```

Comment:
usually one would implement these as before actions/filters.

Reply:
A am actually against using before/after filters. Unless it's something pretty generic done on most requests it does not help the readability of the code.

Reply:
I can understand your concern, but it does protect the developer (e.g. your code will not get executed before all of the other stuff works) to extract an id from a parameter and validate it, or do permissions checks etc are both common and make sense to split across concerns, it also ensure that people have to break down their code so its easier to test, and DRY